command/cat.go

```diff
@@ -0,0 +1,90 @@
+package command
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/peak/s5cmd/storage"
+	"github.com/peak/s5cmd/storage/url"
+)
+
+var catHelpTemplate = `Name:
+	{{.HelpName}} - {{.Usage}}
+
+Usage:
+	{{.HelpName}} [options] source destination
+
+Options:
+	{{range .VisibleFlags}}{{.}}
+	{{end}}
+Examples:
+	1. Print an S3 object's content to stdout
+		 > s5cmd {{.HelpName}} s3://bucket/prefix/object
+`
+
+var catCommandFlags = []cli.Flag{
+	&cli.IntFlag{
+		Name:    "part-size",
+		Aliases: []string{"p"},
+		Value:   defaultPartSize,
+		Usage:   "size of parts transferred between host and remote server, in MiB",
+	},
+}
+
+var CatCommand = &cli.Command{
+	Name:               "cat",
+	HelpName:           "cat",
+	Usage:              "print S3 object's contents to stdout",
+	Flags:              catCommandFlags,
+	CustomHelpTemplate: catHelpTemplate,
+	Before: func(c *cli.Context) error {
+		if c.Args().Len() != 1 {
+			return fmt.Errorf("expected only one argument")
+		}
+
+		src, err := url.New(c.Args().Get(0))
+		if err != nil {
+			return err
+		}
+
+		if src.HasGlob() {
+			return fmt.Errorf("remote source %q can not contain glob characters", src)
+		}
+
+		return nil
+	},
+	Action: func(c *cli.Context) error {
+		src, err := url.New(c.Args().Get(0))
+		if err != nil {
+			return err
+		}
+
+		return Cat(c.Context, src, c.Int64("part-size")*megabytes)
+	},
+}
+
+func Cat(ctx context.Context, src *url.URL, partSize int64) error {
+	client, err := storage.NewClient(src)
+	if err != nil {
+		return err
+	}
+	_, err = client.Get(ctx, src, sequentialWriterAt{w: os.Stdout}, 1, partSize)
```

Comment:
instead of sending actualsize/partSize number of requests sequentially, we can just send a `GetObject` request to download the object in 1 pass. S3 implemention of `Get` should fallback to `GetObject` if given concurrency is 1.

After that, there won't be any need for `part-size` flag.

Reply:
would not `GetObject` load file contents into memory all at once?

Reply:
Why?

Reply:
@igungor  could you check https://github.com/peak/s5cmd/pull/139/commits/5431ab52f69d11f0c3c4e1233146c814ed82dfd8 ?
Did you mean something like that?

Reply:
Do we really need this change or am I missing something? 

```
Specifying a Downloader.Concurrency of 1 will cause the Downloader to download the parts from S3 sequentially.
```

s3manager already checks the given concurrency and makes sequential download if it is 1. I think also giving some dummy part-size (let's say -1) will work, we don't need part-size flag if so.

Reply:
I checked the code and if I'm not mistaken, SDK sends actualSize/partSize number of requests sequentially. So there will be many number of Range requests apart from consuming the object content.

Reply:
Ok. Thanks for the clarification. It is more clear now

Reply:
I guess it's this part
https://github.com/aws/aws-sdk-go/blob/34abf397f4b59e6cae12bb41ea75dcc1d0dc25a9/service/s3/s3manager/download.go#L311-L333